### DISCUSSION TITLE
Identity Map / Unit of Work in MikroOrm v6

### DISCUSSION THREAD

**Discussion by YasiOnFire** [mikro-orm/mikro-orm#5128] [Q&A] [answered]
I've tried to migrate my project to v6 but it's not going great :P
I'm using NestJS and PostgreSQL driver. And I have an `ActivatedGuard` on my controller:
```ts
// ...
const user = await this.userRepository.findOneOrFail(
      { id: userId },
      {
        fields: ['active_account', 'has_2fa_enabled', 'activated', 'onboarded'],
      },
    )
// ...
```

And then in my service i have a `getUsers`  method (which is guarded in the controller) I get the users (simple `this.userRepository.find()`), but my user in the array has only the fields from the guard query and the rest of the array has all the fields. 
![obraz](https://github.com/mikro-orm/mikro-orm/assets/3300701/af741b6e-f67e-4003-a881-0e83bf4515cd)

The same for my `updateUser` route and many others. In v5 I get "what i ask for" so if I ask for specific fields I get specific fields, if i do not provide any i get them all. 
I've read: https://mikro-orm.io/docs/next/guide/first-entity#fork-to-the-win and https://mikro-orm.io/docs/identity-map and I think I understand it, and using a fork works, but what I don't understand is why does the behavior so differ from v5, and why, even tho I see the full select queries I get the data from the first one?

![obraz](https://github.com/mikro-orm/mikro-orm/assets/3300701/7b9bed16-7b27-42ba-9154-10ccb4b91f8a)

I appreciate any hints or suggestions :) Is it a bug or am I missing something?

**Comment by YasiOnFire** [comment]
I've created a reproduction repo:

[![Edit lucid-cache](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/devbox/lucid-cache-r4pnf9?embed=1&file=%2Fsrc%2Fapp.service.ts)


```ts
    const user = await this.usersRepository.findOne(
      {
        id: 1,
      },
      {
        fields: ['email'],
      },
    );

    const user2 = await this.usersRepository.findOne({
      id: 1,
    });

    return {
      user: user,
      user2: user2, // this should have all 4 user properties, no? it does in v5
    };
```

**Reply by B4nan** [reply]
Nope, in your code `user === user2`, no query is even fired, as you query the entity by its PK and it is already loaded. This worked the same since the very beginning, nothing new about it, and it surely works the same in v5.

**Reply by B4nan** [reply]
Actually, the query should be fired (and looks like it correctly is fired), and it should reload the missing fields I think, we nowadays detect whether some entity was only partially loaded (https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L2113). But what I said above still applies, `user === user2`, those two will be serialized the same way if you do it this way.

I am a bit lost in that code sandbox, would be great if you could provide a minimal reproduction - no webservers (no nestjs) involved, same for any other dependency, use just the ORM.

https://github.com/mikro-orm/reproduction

**Reply by YasiOnFire** [reply]
ok, maybe try this:

[![Edit magical-khayyam](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/devbox/magical-khayyam-n8twt6?file=%2Fsrc%2Fexample.test.ts&embed=1)

**Reply by YasiOnFire** [reply]
also here are the exact same test but on v5 (both passing)

[![Edit magical-khayyam (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/devbox/magical-khayyam-forked-lcfm87?embed=1&file=%2Fsrc%2Fexample.test.ts)

**Reply by B4nan** [reply]
thanks, will take a look

**Reply by B4nan** [reply, marked as answer]
Should be fixed by f7b649738a484cbe87c908ca42ec1fa53d963de5.
